server/repository/src/db_diesel/changelog_row.rs

```diff
@@ -67,19 +67,34 @@ impl<'a> ChangelogRowRepository<'a> {
 
     /// # Arguments:
     /// * earliest the first cursor to be included in the returned
-    /// * count the number of entries to be returned
+    /// * limit the number of entries to be returned
     pub fn changelogs(
         &self,
         earliest: u64,
-        count: u32,
+        limit: u32,
```

Comment:
> Thought: I'm not sure if this plays into cargo docs, but could be neat to add the param descriptions on the lines rather than in a comment block above

I agree: 8b36453f9a7ece641f9a24904f10961095722878  in https://github.com/openmsupply/open-msupply/issues/338

Reply:
As for the other comment, why would we want to know if a particular table is tracked by change log, when looking at that table diesel definition ?

Reply:
There is something else I want to do for changelog, i think when you ask for changelog record it should give back the records, not just ids, i'll try to remember to tag you on that PR

Reply:
> As for the other comment, why would we want to know if a particular table is tracked by change log, when looking at that table diesel definition ?

When looking at a table (or well, the struct for the entities that live in it anyway), without contextual knowledge it's not immediately clear whether or not it's central or remote data. In this case it's implicitly remote data because it's included in this list of _server/repository/src/db_diesel/changelog_row.rs_. Anything that isn't in that list is central data, right?

Reminds me a of a gripe in mSupply Desktop Classic™️. There are several similar but unaligned sources of truth for this; Table triggers (if no code then it no sync), sync allowed 2D array (utilised in trigger code, if it exists. Has shortcuts, like you must know that `invoice_line` is child of `invoice`, so in the `invoice_line` trigger pretend it's an `invoice`), central tables array (which isn't all centrally controlled tables, just ones that should go in the change_log!), store tables array (is all store tables... I think), and a table data lookup object (which has some stuff that can indicate being a store record too, but also things that aren't really tables but tricks for doing reverse lookups of certain relations).

Reply:
> When looking at a table (or well, the struct for the entities that live in it anyway), without contextual knowledge it's not immediately clear whether or not it's central or remote data. In this case it's implicitly remote data because it's included in this list of server/repository/src/db_diesel/changelog_row.rs. Anything that isn't in that list is central data, right?

Hmm, I am not sure that change log will be specific to remote data. I was thinking that certain services use change log and do the differentiation on their own, like with sync push, we process change logs and try translate them, it's the defined translators that determine if record is remote data and should be pushed or not. I think this addressed your second paragraph somewhat, one source of truth being defined in the service that uses change log, and the change log is pretty generic.

Reply:
Riiight yea that's cool. Though, does that mean the intent is that all records of all tables should be added to the change log, so that enum `ChangelogTableName` would contain every table in that case?

I'm noticing more curious things 😄 above the enum we `#[derive(DbEnum, Debug, Clone, PartialEq, Eq)]`. Does `DbEnum` indicate to get the table names as per the DB, part of diesel I'm guessing? I thought I'd open vscode and just `ctrl+click` it. This is some arcane stuff beyond me, I can only guess at the moment 😅.